app/audit/AuditType.scala

```diff
@@ -16,23 +16,58 @@
 
 package audit
 
-object AuditType {
-
-  object User {
-    val DepartureDeclarationSubmitted         = "DepartureDeclarationSubmitted"
-    val DepartureCancellationRequestSubmitted = "DepartureCancellationRequestSubmitted"
-  }
-
-  object NCTS {
-    val PositiveAcknowledgementReceived     = "PositiveAcknowledgementReceived"
-    val MrnAllocatedReceived                = "MrnAllocatedReceived"
-    val DeclarationRejectedReceived         = "DeclarationRejectedReceived"
-    val ControlDecisionNotificationReceived = "ControlDecisionNotificationReceived"
-    val NoReleaseForTransitReceived         = "NoReleaseForTransitReceived"
-    val ReleaseForTransitReceived           = "ReleaseForTransitReceived"
-    val CancellationDecisionReceived        = "CancellationDecisionReceived"
-    val WriteOffNotificationReceived        = "WriteOffNotificationReceived"
-    val GuaranteeNotValidReceived           = "GuaranteeNotValidReceived"
-  }
+import models.Enumerable
 
+sealed trait AuditType {}
+
+object AuditType extends Enumerable.Implicits {
+
+  // User
+  case object DepartureDeclarationSubmitted         extends AuditType
+  case object DepartureCancellationRequestSubmitted extends AuditType
+
+  // NCTS
+  case object PositiveAcknowledgementReceived     extends AuditType
+  case object MrnAllocatedReceived                extends AuditType
+  case object DeclarationRejectedReceived         extends AuditType
+  case object ControlDecisionNotificationReceived extends AuditType
+  case object NoReleaseForTransitReceived         extends AuditType
+  case object ReleaseForTransitReceived           extends AuditType
+  case object CancellationDecisionReceived        extends AuditType
+  case object WriteOffNotificationReceived        extends AuditType
+  case object GuaranteeNotValidReceived           extends AuditType
+
+  val values: Seq[AuditType] =
+    Seq(
+      DepartureDeclarationSubmitted,
+      DepartureCancellationRequestSubmitted,
+      PositiveAcknowledgementReceived,
+      MrnAllocatedReceived,
+      DeclarationRejectedReceived,
+      ControlDecisionNotificationReceived,
+      NoReleaseForTransitReceived,
+      ReleaseForTransitReceived,
+      CancellationDecisionReceived,
+      WriteOffNotificationReceived,
+      GuaranteeNotValidReceived
+    )
+
+  val userValues: Seq[AuditType] =
+    Seq(
+      DepartureDeclarationSubmitted,
+      DepartureCancellationRequestSubmitted
+    )
+
+  val nctsValues: Seq[AuditType] =
+    Seq(
+      PositiveAcknowledgementReceived,
+      MrnAllocatedReceived,
+      DeclarationRejectedReceived,
+      ControlDecisionNotificationReceived,
+      NoReleaseForTransitReceived,
+      ReleaseForTransitReceived,
+      CancellationDecisionReceived,
+      WriteOffNotificationReceived,
+      GuaranteeNotValidReceived
+    )
```

Comment:
The userValues and nctsValues variables could be removed as they're not being used.

Reply:
Sure, they were only there in case there was a need to only have a list of those values as I was unsure what other work was going on. Can be easily added back if needed.